src/types.rs

```diff
@@ -136,3 +136,136 @@ impl LintPass for UnitCmp {
         }
     }
 }
+
+pub struct CastPass;
+
+declare_lint!(pub CAST_PRECISION_LOSS, Allow,
+              "casts that cause loss of precision");
+declare_lint!(pub CAST_SIGN_LOSS, Allow,
+              "casts from signed types to unsigned types");
+declare_lint!(pub CAST_POSSIBLE_OVERFLOW, Allow,
+              "casts that may cause overflow");
+
+impl LintPass for CastPass {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(CAST_PRECISION_LOSS,
+                    CAST_SIGN_LOSS,
+                    CAST_POSSIBLE_OVERFLOW)
+    }
+
+    fn check_expr(&mut self, cx: &Context, expr: &Expr) {
+        if let ExprCast(ref ex, _) = expr.node {
+            let (cast_from, cast_to) = (cx.tcx.expr_ty(&*ex), cx.tcx.expr_ty(expr));
+            if cast_from.is_numeric() && !in_external_macro(cx, expr.span) {
```

Comment:
you're only checking `is_numeric` for `cast_from`. I assume that any casts to non-int/float are rejected by the compiler first?

Related question: is there a trait you can `impl` for casts? Might there be in the future?